### DISCUSSION TITLE
Give the weather cards a Dynamic Color tint

### DISCUSSION THREAD

**Discussion by trlef19** [breezy-weather/breezy-weather#1289] [Ideas]
### Describe your idea

In my opinion, weather cards being completely white looks a bit too contrasty and plain so i think it would be nice if they could have an option for a slight Dynamic color tint

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.7 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
There are multiple colored contents inside the cards. These colors, such as icons, are adapted for use with a black or white background. If it's dynamic, technically it means "any color", which we don't support. Btw, we already use little touches of a dynamic tint here and there (title, chart, etc). That would mean rework that as well.

Tl;dr: feel free to try some experimentations, but at this point, I think there would be too much contrast issues.

**Reply by trlef19** [reply]
i meant give it like a very slight tint. Just so it wont be white. I suppose it wouldnt create any problems, right?

**Reply by papjul** [reply]
I don't know if it would fit. You can give it a try and I will review it, but I don't guarantee I will merge it

**Reply by trlef19** [reply]
okay, im not sure if i can do it, i just started learning compose but ill give it a try, can you tell me where the relevant code is?

**Reply by papjul** [reply]
I didn't work on the theming part of the code. Last time I looked into the "pure black" feature asked by some, I spent hours without finding the proper solution, so I can't help you.

The cards on the main screen are not written in Compose, so you will find them amongst the XML files.

**Comment by bugreporter005** [comment]
Second this. Currently, the app only supports global themes (dark & light), but elements like cards don’t always look good with certain weather conditions as shown below:

![weather2](https://github.com/user-attachments/assets/fc5371c2-ece6-47e2-920b-9f7250debef9)
![weather4](https://github.com/user-attachments/assets/cb82ce0f-8072-437f-ba4c-ebf542ecce08)

Therefore, it'd be great to make the cards dynamically adjust their color based on the weather, similar to [Weather Master](https://github.com/PranshulGG/WeatherMaster), for a more visually consistent and pleasant experience.

![weathermaster1](https://github.com/user-attachments/assets/12816882-7cea-4a86-822f-526fcb2d1222)
![weathermaster2](https://github.com/user-attachments/assets/d0b805a0-3296-4d47-ae07-97a6da110b72)

**Reply by papjul** [reply]
Go to Settings > Appearance.
Turn on Day/Night mode for locations.

**Reply by bugreporter005** [reply]
> Day/Night mode for locations.

What I meant was two global themes alone don’t fit all weather conditions as the contrast gap between the background and cards is still too much and makes the app look a bit outdated by today's design standards. One solution could be mimicking Material You by extracting card colors from the background and making them slightly darker for better contrast. Here's a simple mockup to demonstrate the idea (you're welcome to further tune the colors):

![20250305_015312_015353](https://github.com/user-attachments/assets/2e2f6025-94f2-4155-a143-70d423d5e414)
![20250305_015135_015356](https://github.com/user-attachments/assets/c512816f-499c-4175-aecc-c6048ac15841)

**Reply by papjul** [reply]
Oh, I see what you want to do now.
Problem again is contrast.
It works for the app you give in example because they use subcards inside the card.
I will try some tests, but if it ever matches, the tint will be way more subtle.
